spec/CachePluginSpec.php

```diff
@@ -97,6 +97,49 @@ function it_doesnt_store_post_requests(CacheItemPoolInterface $pool, CacheItemIn
         $this->handleRequest($request, $next, function () {});
     }
 
+    function it_stores_post_requests_when_allowed(CacheItemPoolInterface $pool, CacheItemInterface $item, RequestInterface $request, ResponseInterface $response, StreamFactory $streamFactory, StreamInterface $stream)
+    {
+        $this->beConstructedWith($pool, $streamFactory, [
+            'default_ttl' => 60,
+            'cache_lifetime' => 1000,
+            'methods' => ['GET', 'HEAD', 'POST']
+        ]);
+
+        $httpBody = 'hello=world';
+        $stream->__toString()->willReturn($httpBody);
+        $stream->isSeekable()->willReturn(true);
+        $stream->rewind()->shouldBeCalled();
+
+        $request->getMethod()->willReturn('POST');
+        $request->getUri()->willReturn('/post');
+        $request->getBody()->willReturn($stream);
+
+        $response->getStatusCode()->willReturn(200);
+        $response->getBody()->willReturn($stream);
+        $response->getHeader('Cache-Control')->willReturn(array())->shouldBeCalled();
```

Comment:
Please use short array syntax (StyleCI is turned off for spec files because of the snake_case).